include/taco/tensor.h

```diff
@@ -53,10 +53,10 @@ class TensorBase {
   /// Get the order of the tensor (the number of modes).
   size_t getOrder() const;
 
-  /// Get the size of a tensor mode.
-  int getDimension(size_t dim) const;
+  /// Get the dimension of a tensor mode.
+  int getDimension(size_t mode) const;
 
-  /// Get a vector with the size of each tensor mode.
+  /// Get a vector with the dimension of each tensor mode.
```

Comment:
I think this change makes it harder to understand what is going on. Not everyone will be that fluent at the mode/dimension difference. Since dimension == size of mode, I think this comment should stay as it was.

Reply:
I was going for consistency, so perhaps there are a few areas where it would be easier for users to see size instead of dimension. I was worried that the "size" of the tensor mode would be interpreted as the memory size of the mode index. I do intend on including a glossary in a later commit which explains what all the terms mean.

Reply:
In our glossary, we should also include equivalent terms from other languages. Like a mode is called an axis in numpy

Reply:
The glossary is a good idea.

The byte size confusion is a good point. When we have the glossary we could also put a parenthesis in this spot with the definition, e.g., "dimension (size of tensor mode in number of components)".

Where does the glossary go? Presumably in the documentation of the tensor.h or the class Tensor? We should also add it to the documentation on the web page.